api/src/main/java/ai/djl/util/Resource.java

```diff
@@ -16,15 +16,17 @@
 import java.util.concurrent.atomic.AtomicReference;
 
 /**
- * {@code NativeResource} is an internal class for {@link AutoCloseable} blocks of memory created in
- * the different engines.
+ * {@code Resource} is an internal class for {@link AutoCloseable} blocks of memory created in the
+ * different engines.
+ *
+ * @param <T> the resource that could map to a native pointer or java object
  */
-public abstract class NativeResource<T> implements AutoCloseable {
+public abstract class Resource<T> implements AutoCloseable {
```

Comment:
Why we rename this class?

Resource seems to general. And we have a Resource class in repository package.